tests/ui/vec.rs

```diff
@@ -176,3 +176,38 @@ fn below() {
         let _: String = a;
     }
 }
+
+fn func_needing_vec(_bar: usize, _baz: Vec<usize>) {}
+fn func_not_needing_vec(_bar: usize, _baz: usize) {}
+
+fn issue11861() {
+    macro_rules! this_macro_needs_vec {
+        ($x:expr) => {{
+            func_needing_vec($x.iter().sum(), $x);
+            for _ in $x {}
+        }};
+    }
+    macro_rules! this_macro_doesnt_need_vec {
+        ($x:expr) => {{ func_not_needing_vec($x.iter().sum(), $x.iter().sum()) }};
+    }
+
+    // Do not lint the next line
+    this_macro_needs_vec!(vec![1]);
+    // Lint the next line
+    this_macro_doesnt_need_vec!(vec![1]);
```

Comment:
You can use headers like `//~ ERROR` for a standardized form of error marking (then, `cargo test` will check for these)

```suggestion
    this_macro_doesnt_need_vec!(vec![1]); //~ ERROR: <lint message>
```

Or, in a different line

```suggestion
    this_macro_doesnt_need_vec!(vec![1]);
    //~^ ERROR: <lint message>
```

Same with all the other occurrences in this test file